### PR TITLE
Document bundled extension format

### LINKED ISSUE #6855: Document how to bundle extensions with the new extension loading process
**What would you like to be added**:
Explain/document how to include/bundle extensions at build time of the OpenLens/Lens Core app itself.

**Why is this needed**:
Now that the bundled extensions have been removed from this repository, there is no documentation available, either in prose or code, about how to include an extension as part of the application bundle itself. This includes details such as:
- Where in the build tree one would put bundled extensions in order to have the build system pick them up
- What changes, if any, are needed in `package.json` or other configuration files
- What changes, if any, are needed to the extension to be bundled that may be different from the "usual" extension format/anatomy as described at https://api-docs.k8slens.dev/v6.0.1.

On another note, the docs at that link don't appear to have been updated for a while. They should be updated to ensure that vendors and the community can continue to produce and maintain extensions that are compatible and provide the best possible experience to end users of Lens Desktop and other Lens Core-based applications.

**Environment you are Lens application on:**

- Kubernetes distribution: GKE/k3s
- Desktop OS: macOS

### PR DESCRIPTION
**Pull request by Nokel81**
- Bundled extensions are always enabled, and are always compatible
- Have bundled extensions be loaded asyncronously to support
  typescript dynamic import (which is typed) as opposed to require

Signed-off-by: Sebastian Malton <sebastian@malton.name>

resolves #6855

### CODE REVIEW

File: packages/sample-extension/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Description
+
+This is a CommonJS webpack transpilation of the `node-fetch` package for use within `electron-renderer`.
```
Comment (jakolehm) [inline packages/sample-extension/README.md]: Please update to match sample-extension.

**Review by jakolehm — COMMENTED** [review]
IMHO all changes to extension discovery/loader should be in a separate PR.

File: packages/sample-extension/README.md (hunk 1 above)
Comment (Nokel81) [reply packages/sample-extension/README.md]: Sounds good

**Comment by Nokel81** [comment]
> IMHO all changes to extension discovery/loader should be in a separate PR.

Sure will do

File: packages/sample-extension/README.md (hunk 1 above)
Comment (jim-docker) [reply packages/sample-extension/README.md]: This is resolved? This sample just logs when the extension is activated and deactivated in the renderer process(es)

File: packages/open-lens/src/common/sample-bundled-extension.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { getInjectable } from "@ogre-tools/injectable";
+import { bundledExtensionInjectionToken } from "@k8slens/core/common";
+import manifest from "@k8slens/sample-extension/package.json";
+
+const cliLensExtensionInjectable = getInjectable({
```
Comment (jim-docker) [inline packages/open-lens/src/common/sample-bundled-extension.injectable.ts]:
```suggestion
const sampleBundledExtensionInjectable = getInjectable({
```
similar for id and export

**Comment by jim-docker** [comment]
Is there something else I have to do first?

```
~/git/lensapp/lens/packages/open-lens$ npm install
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@k8slens%2fsample-extension - Not found
npm ERR! 404 
npm ERR! 404  '@k8slens/sample-extension@^6.4.0-beta.13' is not in this registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/user/.npm/_logs/2023-02-16T22_15_24_067Z-debug.log
```

**Comment by Nokel81** [comment]
Yes I would would run `yarn install` from the root of the repo instead.

**Comment by jim-docker** [comment]
`yarn install` ran without errors but `npm run dev` or `yarn run dev` result in this (run from the repo root or `packages/open-lens`-- what is the right way to start dev OpenLens?):

```
[renderer] ERROR in ./src/common/sample-bundled-extension.injectable.ts 10:33-73
[renderer] Module not found: Error: Can't resolve '@k8slens/sample-extension/main' in '/Users/user/git/lensapp/lens/packages/open-lens/src/common'
```

**Pull request closed without merging**